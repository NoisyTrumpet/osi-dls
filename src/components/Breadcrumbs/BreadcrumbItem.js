import React from 'react';
import PropTypes from 'prop-types';
import { List } from '../List';
import { Typography } from '../Typography';

const BreadcrumbItem = ({
  to,
  title,
  target,
  rel,
}) => {
  const typography = <Typography variant="breadcrumb">{title}</Typography>;
  return (
    <List.Item>
      {to
        ? (
          <a href={to} className="breadcrumb-link" target={target} rel={rel}>
            {typography}
          </a>
        )
        : typography}
    </List.Item>
  );
};

BreadcrumbItem.defaultProps = {};

BreadcrumbItem.propTypes = {
  title: PropTypes.string.isRequired,
  to: PropTypes.string,
  target: PropTypes.string,
  rel: PropTypes.string,
};

export default BreadcrumbItem;
