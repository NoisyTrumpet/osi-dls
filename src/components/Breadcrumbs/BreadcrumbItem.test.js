import '@testing-library/jest-dom/extend-expect'; // grants us all jest-dom methods
import React from 'react';
import { render, screen } from '@testing-library/react';
import BreadcrumbItem from './BreadcrumbItem';

const breadcrumbsData = [
  {
    title: 'Page1',
    to: '#somewhere',
  },
  {
    title: 'Page2',
    to: '#somewhereElse',
  },
  {
    title: 'Page3',
    to: '#finalDestination',
  },
];

describe('BreadcrumbItems component', () => {
  test('should render the correct title', () => {
    const { rerender } = render(
      <BreadcrumbItem title={breadcrumbsData[0].title} />,
    );

    expect(screen.getByText(breadcrumbsData[0].title)).toBeInTheDocument();

    rerender(<BreadcrumbItem title={breadcrumbsData[1].title} />);
    expect(screen.getByText(breadcrumbsData[1].title)).toBeInTheDocument();
    expect(screen.queryByText(breadcrumbsData[0].title)).toBeNull();
  });

  test('should render an anchor with the right target', () => {
    const { rerender } = render(
      <BreadcrumbItem
        title={breadcrumbsData[0].title}
        to={breadcrumbsData[0].to}
        target="_blank"
      />,
    );

    expect(screen.getByRole('link')).toHaveAttribute('target', '_blank');

    rerender(
      <BreadcrumbItem
        title={breadcrumbsData[0].title}
        to={breadcrumbsData[0].to}
      />,
    );
    expect(screen.getByRole('link')).not.toHaveAttribute('target');
  });

  test('should render an anchor with the right rel', () => {
    const { rerender } = render(
      <BreadcrumbItem
        title={breadcrumbsData[0].title}
        to={breadcrumbsData[0].to}
        target="_blank"
        rel="noopener"
      />,
    );

    expect(screen.getByRole('link')).toHaveAttribute('rel', 'noopener');

    rerender(
      <BreadcrumbItem
        title={breadcrumbsData[0].title}
        to={breadcrumbsData[0].to}
        target="_blank"
      />,
    );
    expect(screen.getByRole('link')).not.toHaveAttribute('rel');
  });

  test('matches previous snapshot', () => {
    const { container } = render(
      <BreadcrumbItem
        title={breadcrumbsData[0].title}
        to={breadcrumbsData[0].to}
        target="_blank"
        rel="noopener"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
