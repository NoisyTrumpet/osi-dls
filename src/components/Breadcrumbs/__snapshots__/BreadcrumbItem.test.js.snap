// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbItems component matches previous snapshot 1`] = `
<div>
  <li
    class="list-item"
  >
    <a
      class="breadcrumb-link"
      href="#somewhere"
      rel="noopener"
      target="_blank"
    >
      <span
        class="typography typography-breadcrumb"
      >
        Page1
      </span>
    </a>
  </li>
</div>
`;
